### PR TITLE
MOSIP-45312 - Automated esignet-signup test cases

### PR DESCRIPTION
**Pull request by prathmeshj12**
MOSIP-45312 - Automated esignet-signup test cases

## Summary by CodeRabbit

- **New Features**
  - Added validation for missing response attributes.
  - Added cookie-expiry verification based on token validity.
  - Added configurable WebSocket content types, alternate destinations, concurrent subscriptions, and abnormal-close handling.

- **Tests**
  - Expanded signup, authentication, OAuth, identity-verification, claim-details, and UI-spec coverage.
  - Added scenarios for duplicate frames, rapid messaging, invalid content types, failed statuses, unrecognized slots, step details, rejected subscriptions, and abnormal WebSocket closures.

### CODE REVIEW

File: api-test/src/main/java/io/mosip/testrig/apirig/signup/testscripts/SimplePostForAutoGenId.java [hunk 1]
```diff
@@ -212,9 +218,31 @@ public void test(TestCaseDTO testCaseDTO)
...
+					"Decoded ID token is missing exp/iat claims; cannot verify cookie expiry");
+		}
+		long tokenValiditySeconds = payload.getLong("exp") - payload.getLong("iat");
+
+		List<Cookie> cookies = response.getDetailedCookies().asList();
+		if (cookies.isEmpty()) {
+			throw new AdminTestException(
+					"Expected a Set-Cookie header on the prepare-signup-redirect response, but none was found");
+		}
+		long cookieMaxAge = cookies.get(0).getMaxAge();
```
[the remark this answers is not in the document]
Comment (prathmeshj12) [reply api-test/src/main/java/io/mosip/testrig/apirig/signup/testscripts/SimplePostForAutoGenId.java]: COOKIENAME doesn't exist anywhere in this codebase (your own analysis script found nothing for it either). Traced the actual prepare-signup-redirect implementation in AuthorizationServiceImpl#prepareSignupRedirect: the cookie is named after the transaction ID itself (new Cookie(signupRedirectRequest.getTransactionId(), ...)), not a fixed name. Fixed by pulling transactionId from the response body and selecting the cookie by that name instead of cookies.get(0).

File: api-test/src/main/java/io/mosip/testrig/apirig/signup/testscripts/WebSocketConnection.java [hunk 2]
```diff
@@ -344,10 +397,106 @@ private void assertSlotReconnectionRejected(SignupCustomWebSocketClientUtil firs
...
+	}
+
+	private String collectErrorFrame() {
+		for (Map.Entry<String, String> entry : SignupCustomWebSocketClientUtil.getMessageStore().entrySet()) {
+			if (entry.getKey() != null && entry.getKey().startsWith("ERROR-")) {
+				return entry.getValue();
+			}
+		}
+		return null;
+	}
```
[the remark this answers is not in the document]
Comment (prathmeshj12) [reply api-test/src/main/java/io/mosip/testrig/apirig/signup/testscripts/WebSocketConnection.java:501]: Fixed as suggested — now logs the full exception via logger.error(...) before throwing, and includes e (not just e.getMessage()) in the thrown message so the exception type/context isn't lost.

File: api-test/src/main/java/io/mosip/testrig/apirig/signup/testscripts/WebSocketConnection.java [hunk 3]
```diff
@@ -344,10 +397,106 @@ private void assertSlotReconnectionRejected(SignupCustomWebSocketClientUtil firs
...
+		if (wsSession == null) {
+			throw new AdminTestException("Expected the WebSocket connection to be established, but it was not");
+		}
+
+		try {
+			wsSession.close(new CloseReason(CloseReason.CloseCodes.GOING_AWAY, "Simulated abnormal disconnect"));
+		} catch (Exception e) {
+			throw new AdminTestException(
+					"Failed to close the websocket session with an abnormal close code: " + e.getMessage());
+		}
```
[the remark this answers is not in the document]
Comment (prathmeshj12) [reply api-test/src/main/java/io/mosip/testrig/apirig/signup/testscripts/WebSocketConnection.java:444]: Fixed — added clearErrorFrames(), called right before connect() whenever expectSubscriptionRejected is set, so any leftover ERROR- entry from a prior test run can't be mistaken for this attempt's own subscription rejection.

File: api-test/src/main/resources/config/testCaseInterDependency_mock.json [hunk 4]
```diff
@@ -779,10 +879,22 @@
...
+        "TC_Signup_ESignet_OAuthDetailsRequestV3_Prerequisite_27"
+    ],
+    "TC_ESignet_SignupAuthorize_18": [
+        "TC_Signup_ESignet_OAuthDetailsRequestV3_Prerequisite_23",
+        "TC_Signup_ESignet_PrepareSignupRedirect_Prerequisite_19"
+    ],
     "TC_ESignet_VerifyChallengeForResetPasswordNegTC_15": [
         "TC_ESignet_VerifyChallengeNegTC_40",
         "TC_ESignet_VerifyChallengeForResetPasswordNegTC_15"
     ],
```
[the remark this answers is not in the document]
Comment (prathmeshj12) [reply api-test/src/main/resources/config/testCaseInterDependency_mock.json:893]: Checked — this self-reference is pre-existing upstream data, not something this PR introduced (confirmed via git diff against upstream/develop: that block is unchanged context, not part of our diff). It's also not unique to _15 — _10 through _17 all have the identical self-reference bug, so it looks like a systemic issue in this auto-generated file rather than a one-off. I don't have confident visibility into the correct intended prerequisite for any of them, and fixing just this one instance while leaving the identical bug in six siblings would be inconsistent. Flagging this as a separate cleanup rather than guess-fixing it here.

File: api-test/src/main/java/io/mosip/testrig/apirig/signup/testscripts/WebSocketConnection.java [hunk 5]
```diff
@@ -131,9 +124,11 @@ public void test(TestCaseDTO testCaseDTO)
 		tempUrl = tempUrl.replace("https", "wss") + testCaseDTO.getEndPoint() + "?slotId=" + slotId;
 
 		SignupCustomWebSocketClientUtil webSocketClient = new SignupCustomWebSocketClientUtil(cookie, subscribeDestination, sendDestination);
-		
 
-		// Connect to WebSocket server
+		if (expectSubscriptionRejected) {
+			clearErrorFrames();
+		}
+
```
[the remark this answers is not in the document]
Comment (prathmeshj12) [reply api-test/src/main/java/io/mosip/testrig/apirig/signup/testscripts/WebSocketConnection.java:131]: This suite runs strictly sequentially — no parallel attribute is set anywhere in testNgXmlFiles/*.xml, and this @Test has no threadPoolSize/invocationCount override. TestNG's default (sequential) execution means clearErrorFrames() → connect() → pollErrorFrame() in assertSubscriptionRejected() all run inside one synchronous test-method call, with no other test method executing concurrently in between — so the "overlapping tests" scenario described here (two tests' subscription attempts racing each other) doesn't apply to this suite as configured.

There is a narrower, real edge case this doesn't cover: onMessage fires on the WebSocket container's own I/O thread, asynchronously from the TestNG thread, so a slow-to-close previous connection could theoretically deliver a late frame during the next test's brief clear-to-poll window. Fixing that properly would mean scoping clearErrorFrames()/pollErrorFrame() by subscribeDestination, but that requires confirming the actual STOMP ERROR frame body/headers contain a destination reference — I haven't verified that against a live captured frame, so I'm not comfortable guess-implementing filtering logic that might silently match nothing. Leaving as-is for now; flagging this as a known, narrow edge case rather than fixing speculatively.

**Review by damodarguru — APPROVED** [review]

**Review by mohanachandran-s — APPROVED** [review]
